File: ts/project/.sdk/tm/rb/feature/base_feature.rb
# ProjectName SDK base feature

class ProjectNameBaseFeature
  attr_accessor :version, :name, :active

  # Positions this feature when added via the client `extend` option:
  # "__before__" / "__after__" / "__replace__" name an already-added
  # feature (mirrors the ts feature `_options`).
  attr_accessor :_options

  def initialize
    @version = "0.0.1"
    @name = "base"
    @active = true
  end

  def get_version; @version; end
  def get_name; @name; end
  def get_active; @active; end

  def init(ctx, options); end
  def PostConstruct(ctx); end
  def PostConstructEntity(ctx); end
  def SetData(ctx); end
  def GetData(ctx); end
  def GetMatch(ctx); end
  def SetMatch(ctx); end
  def PrePoint(ctx); end
  def PreSpec(ctx); end
  def PreRequest(ctx); end
  def PreResponse(ctx); end
  def PreResult(ctx); end
  def PreDone(ctx); end
  def PreUnexpected(ctx); end
end

File: ts/project/.sdk/tm/py/feature/base_feature.py
# ProjectName SDK base feature

from __future__ import annotations


class ProjectNameBaseFeature:
    def __init__(self):
        self.version = "0.0.1"
        self.name = "base"
        self.active = True

    def get_version(self):
        return self.version

    def get_name(self):
        return self.name

    def get_active(self):
        return self.active

    def init(self, ctx, options):
        pass

    def PostConstruct(self, ctx):
        pass

    def PostConstructEntity(self, ctx):
        pass

    def SetData(self, ctx):
        pass

    def GetData(self, ctx):
        pass

    def GetMatch(self, ctx):
        pass

    def SetMatch(self, ctx):
        pass

    def PrePoint(self, ctx):
        pass

    def PreSpec(self, ctx):
        pass

    def PreRequest(self, ctx):
        pass

    def PreResponse(self, ctx):
        pass

    def PreResult(self, ctx):
        pass

    def PreDone(self, ctx):
        pass

    def PreUnexpected(self, ctx):
        pass
